src/main/java/games/strategy/engine/GameEngineUtils.java

```diff
@@ -0,0 +1,52 @@
+package games.strategy.engine;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import java.util.Comparator;
+
+import games.strategy.util.Version;
+
+/**
+ * A collection of methods for use with game engines.
+ */
+public final class GameEngineUtils {
+  private GameEngineUtils() {}
+
+  /**
+   * Indicates the engines with the specified versions are compatible.
+   *
+   * @param engineVersion The first engine version.
+   * @param otherEngineVersion The second engine version.
+   *
+   * @return {@code true} if the engines are compatible; otherwise {@code false}.
+   */
+  public static boolean isEngineCompatibleWithEngine(final Version engineVersion, final Version otherEngineVersion) {
```

Comment:
These methods were originally extracted because the definition of engine-engine compatibility and engine-map compatibility was buried in the `Version` class.  `Version` is used throughout the code to represent many different types of versions (e.g. engine, map, lobby protocol, etc.).  It wasn't clear why we were promoting functionality specific to engine versions in what should just be a simple value type representing a four-component version.

I agree that there is probably a leaky implementation here regarding the comparison.  If you go back to f4ca9dd, you'll see that it was still encapsulated within `Version` at the expense of providing an option to ignore the micro version during comparison.  I think the argument for getting rid of that option was that it was yet another implementation detail that had to do with some specific use cases (e.g. engine-engine compatibility) and had nothing to do with the general `Version` functionality.

Here's another way we could address these competing forces: provide an method on `Version` that returns the instance without the micro version:

```java
public Version withoutMicro() {
  return new Version(m_major, m_minor, m_point, 0);
}
```

Then the comparison logic within `Version` doesn't need to be duplicated outside that class and their need be no special overloads that accept options for ignoring certain fields.  For example:

```java
engineVersion.withoutMicro().compareTo(otherEngineVersion.withoutMicro()) == 0
```

Actually, with this approach, you can just re-use `equals()` in this situation (which we couldn't do before with any of the previous approaches):

```java
engineVersion.withoutMicro().equals(otherEngineVersion.withoutMicro())
```

Reply:
The suggestion here sounds good :+1: